src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelBase.cs

```diff
@@ -204,7 +214,43 @@ protected ViewModelBase()
         /// <exception cref="ModelNotRegisteredException">A mapped model is not registered.</exception>
         /// <exception cref="PropertyNotFoundInModelException">A mapped model property is not found.</exception>
         protected ViewModelBase(bool supportIEditableObject, bool ignoreMultipleModelsWarning = false, bool skipViewModelAttributesInitialization = false)
-            : this(null, supportIEditableObject, ignoreMultipleModelsWarning, skipViewModelAttributesInitialization)
+            : this(false, supportIEditableObject, ignoreMultipleModelsWarning, skipViewModelAttributesInitialization)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ViewModelBase"/> class.
+        /// </summary>
+        /// <param name="reuseIdentifier">
+        /// if set to <c>true</c>, the identifier will be reused from the relased identifier pool.
+        /// </param>
+        /// <param name="supportIEditableObject">if set to <c>true</c>, the view model will natively support models that
+        /// implement the <see cref="IEditableObject"/> interface.</param>
+        /// <param name="ignoreMultipleModelsWarning">if set to <c>true</c>, the warning when using multiple models is ignored.</param>
+        /// <param name="skipViewModelAttributesInitialization">
+        /// if set to <c>true</c>, the initialization will be skipped and must be done manually via <see cref="InitializeViewModelAttributes"/>.
+        /// </param>
+        /// <exception cref="ModelNotRegisteredException">A mapped model is not registered.</exception>
+        /// <exception cref="PropertyNotFoundInModelException">A mapped model property is not found.</exception>
+        protected ViewModelBase(bool reuseIdentifier, bool supportIEditableObject, bool ignoreMultipleModelsWarning = false, bool skipViewModelAttributesInitialization = false)
```

Comment:
Not sure whether this is the way to go. Since we are adding the bool in front, and we use default values, the compiler *could* use this instructor while the user wanted to use the above one. Order of parameters of overloads should always be the same for each, so we need to add it to the end (with a default value of false). Yes, the compiler might complain that it doesn't know which one to pick, not sure how to solve that one (yet). The question is: is it really that important to allow re-use of id's for vm's?

@GreenKn1ght were you running out of id's for the vm's or for other objects?

Reply:
For VMs.

Reply:
@GeertvanHorrik I keep the old constructor (so, no breaking change at this point) and just added a new one with the parameter in front.  The reason is exactly like you described before.

Reply:
I believe it's not the same. If I call base(true), which one gets picked?

Reply:
will pick this one 

        protected ViewModelBase(bool supportIEditableObject, bool ignoreMultipleModelsWarning = false, bool skipViewModelAttributesInitialization = false)
            : this(false, supportIEditableObject, ignoreMultipleModelsWarning, skipViewModelAttributesInitialization)
        {
        }

notice that there is no default value in the new one for supportIEditableObject

        protected ViewModelBase(bool reuseIdentifier, bool supportIEditableObject,...

Reply:
And what will happen if I do:

: base(true, true)